Generate serialization for DocumentEditingContext

sheeparegreat opened the pull request:
#### ed1bd1226f001dbc53bfa2b0ca56aac0db859258
<pre>
Generate serialization for DocumentEditingContext
<a href="https://bugs.webkit.org/show_bug.cgi?id=262821">https://bugs.webkit.org/show_bug.cgi?id=262821</a>
rdar://116603017

Reviewed by Ryosuke Niwa.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/DocumentEditingContext.h:
* Source/WebKit/Shared/DocumentEditingContext.mm:
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContext::Range&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContext::Range&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContext::TextRectAndRange&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContext::TextRectAndRange&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContext&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContext&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContextRequest&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::DocumentEditingContextRequest&gt;::decode): Deleted.
* Source/WebKit/Shared/DocumentEditingContext.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/269112@main">https://commits.webkit.org/269112@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4e771d6801122bc3a3533773bfb8db9f62644e3e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21573 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/21894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22625 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23437 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19986 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/25187 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22123 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21173 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21800 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21438 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18703 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24288 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18639 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25853 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19702 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19762 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23703 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20242 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17251 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19562 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23798 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2678 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20147 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/Shared/DocumentEditingContext.serialization.in:
```diff
@@ -0,0 +1,78 @@
+
+[Nested] struct WebKit::DocumentEditingContext::Range {
+    uint64_t location;
+    uint64_t length;
```
We should probably add a validator that location + length doesn't overflow:
e.g. [Validator='!(Checked<uint64_t> { *location } + *length).hasOverflowed()'] length.

sheeparegreat replied:
Rebased commit with main and added the suggested validator.

sheeparegreat committed:
Generate serialization for DocumentEditingContext
https://bugs.webkit.org/show_bug.cgi?id=262821
rdar://116603017

Reviewed by Ryosuke Niwa.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/DocumentEditingContext.h:
* Source/WebKit/Shared/DocumentEditingContext.mm:
(IPC::ArgumentCoder<WebKit::DocumentEditingContext::Range>::encode): Deleted.
(IPC::ArgumentCoder<WebKit::DocumentEditingContext::Range>::decode): Deleted.
(IPC::ArgumentCoder<WebKit::DocumentEditingContext::TextRectAndRange>::encode): Deleted.
(IPC::ArgumentCoder<WebKit::DocumentEditingContext::TextRectAndRange>::decode): Deleted.
(IPC::ArgumentCoder<WebKit::DocumentEditingContext>::encode): Deleted.
(IPC::ArgumentCoder<WebKit::DocumentEditingContext>::decode): Deleted.
(IPC::ArgumentCoder<WebKit::DocumentEditingContextRequest>::encode): Deleted.
(IPC::ArgumentCoder<WebKit::DocumentEditingContextRequest>::decode): Deleted.
* Source/WebKit/Shared/DocumentEditingContext.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/269112@main
